soh/src/code/z_message_PAL.c

```diff
@@ -1682,6 +1682,12 @@ void Message_OpenText(GlobalContext* globalCtx, u16 textId) {
         memcpy(font->msgBuf, src, font->msgLength);
     }
 
+    // (From z_player.c) In rando, Navi gives rando-related advice
+    if (textId == 0x0140 && gSaveContext.n64ddFlag) {
+        RandoNaviTip(globalCtx);
+        msgCtx->msgLength = font->msgLength = strlen(font->msgBuf);
+    }
+
```

Comment:
For the custom-message system, you'll want to move this check to be in the function `CustomMessage_RetrieveIfExists` which is in `OTRGlobals.cpp`, and change it to retrieve a message from a custom message table.